### PR TITLE
add is_double_vote and is_surround_vote

### PR DESCRIPTION
**Pull request by thojest**
## Issue Addressed

#207 - Implement double and surround vote functions

## Proposed Changes

Implementing these vote functions from the phase0-spec:

- https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#is_double_vote

- https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#is_surround_vote

### CODE REVIEW

**Comment by paulhauner** [comment]
Thanks @thojest!

We're just fixing a failing test on master, then we'll run these tests again. They're not failing because of you (at this point).

**Comment by paulhauner** [comment]
Re-running tests now :)

**Comment by paulhauner** [comment]
@thojest can you please merge the latest master onto this PR :)

Otherwise the tests wont pass.

**Comment by thojest** [comment]
@paulhauner done :)

File: eth2/types/src/slashable_vote_data.rs [hunk 1]
```diff
@@ -13,6 +14,21 @@ pub struct SlashableVoteData {
     pub aggregate_signature: AggregateSignature,
 }
 
+impl SlashableVoteData {
+    pub fn is_double_vote(&self, other: &SlashableVoteData, spec: &ChainSpec) -> bool {
```
[the remark this answers is not in the document]
Comment (thojest) [reply eth2/types/src/slashable_vote_data.rs]: agree

File: eth2/types/src/slashable_vote_data.rs [hunk 2]
```diff
@@ -13,6 +14,21 @@ pub struct SlashableVoteData {
...
+        self.data.slot.epoch(spec.epoch_length) == other.data.slot.epoch(spec.epoch_length)
+    }
+
+    pub fn is_surround_vote(&self, other: &SlashableVoteData, spec: &ChainSpec) -> bool {
```
[the remark this answers is not in the document]
Comment (thojest) [reply eth2/types/src/slashable_vote_data.rs]: agree :+1:

File: eth2/types/src/slashable_vote_data.rs [hunk 3]
```diff
@@ -66,9 +82,71 @@ impl<T: RngCore> TestRandom<T> for SlashableVoteData {
...
+            slashable_vote_first.is_double_vote(&slashable_vote_second, &spec),
+            false
+        );
+    }
+
+    #[test]
+    pub fn test_is_surround_vote_true() {
+        let spec = ChainSpec::foundation();
+        let slashable_vote_first = create_slashable_vote_data(2, 1, &spec);
+        let slashable_vote_second = create_slashable_vote_data(1, 2, &spec);
```
[the remark this answers is not in the document]
Comment (thojest) [reply eth2/types/src/slashable_vote_data.rs]: thx for the feedback on this one. i will add the realistic test scenario. :+1:

**Comment by paulhauner** [comment]
Great work, merging now!

**Pull request merged**